packages/backend-ai/src/types.ts

```diff
@@ -0,0 +1,197 @@
+export type JSONLike =
+  | string
+  | number
+  | boolean
+  | null
+  | { [key: string]: JSONLike }
+  | JSONLike[];
+
+export type JSONSchema = {
+  type: string;
+  properties?: Record<string, JSONSchema>;
+  required?: string[];
+  items?: JSONSchema;
+  enum?: (string | number | boolean | null)[];
+  minimum?: number;
+  maximum?: number;
+  minLength?: number;
+  maxLength?: number;
+  pattern?: string;
+  format?: string;
+  description?: string;
+  default?: JSONLike;
+  additionalProperties?: boolean | JSONSchema;
+};
+
+export type Prompt = {
+  text: string;
+};
+
+export type ContentBlock =
+  | TextContent
+  | ImageContent
+  | DocumentContent
+  | ToolUseContent
+  | ToolResultContent
+  | GuardContentContent
+  | $UnknownContent;
+
+export type TextContent = {
+  text: string;
+  image?: never;
+  document?: never;
+  toolUse?: never;
+  toolResult?: never;
+  guardContent?: never;
+  $unknown?: never;
+};
+
+export type ImageContent = {
+  text?: never;
+  image: ImageBlock;
+  document?: never;
+  toolUse?: never;
+  toolResult?: never;
+  guardContent?: never;
+  $unknown?: never;
+};
+
+export type DocumentContent = {
+  text?: never;
+  image?: never;
+  document: DocumentBlock;
+  toolUse?: never;
+  toolResult?: never;
+  guardContent?: never;
+  $unknown?: never;
+};
+
+export type ToolUseContent = {
+  text?: never;
+  image?: never;
+  document?: never;
+  toolUse: ToolUseBlock;
+  toolResult?: never;
+  guardContent?: never;
+  $unknown?: never;
+};
+
+export type ToolResultContent = {
+  text?: never;
+  image?: never;
+  document?: never;
+  toolUse?: never;
+  toolResult: ToolResultBlock;
+  guardContent?: never;
+  $unknown?: never;
+};
+
+export type GuardContentContent = {
+  text?: never;
+  image?: never;
+  document?: never;
+  toolUse?: never;
+  toolResult?: never;
+  guardContent: GuardrailConverseContentBlock;
+  $unknown?: never;
+};
+export type $UnknownContent = {
+  text?: never;
+  image?: never;
+  document?: never;
+  toolUse?: never;
+  toolResult?: never;
+  guardContent?: never;
+  $unknown?: [string];
+};
+
+export type ImageBlock = {
+  format: 'png' | 'jpeg' | 'gif' | 'webp'; // eslint-disable-line spellcheck/spell-checker
+  source: {
+    bytes: Uint8Array;
```

Comment:
Talking with the Bedrock team to see if we can pass a base64 string because that will be easier for us